libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt

```diff
@@ -179,18 +179,18 @@ class RustMatrixClient constructor(
         visibleRoomsSlidingSyncList.destroy()
         slidingSync.destroy()
         verificationService.destroy()
+        client.destroy()
     }
 
     override suspend fun logout() = withContext(dispatchers.io) {
-        close()
         try {
             client.logout()
         } catch (failure: Throwable) {
             Timber.e(failure, "Fail to call logout on HS. Still delete local files.")
         }
         baseDirectory.deleteSessionDirectory(userID = client.userId())
         sessionStore.removeSession(client.userId())
-        client.destroy()
+        close()
```

Comment:
Not blocking (and not new), but is it the responsibility to the class to close itself?

Reply:
I'd say when it's a logout, yes, but both are valid options